packages/app/src/cli/services/function/replay.test.ts

```diff
@@ -304,6 +326,6 @@ function expectExecToBeCalledWithInput(input: any) {
 }
 
 function mockFileOperations(data: {run: FunctionRunData; path: string}[]) {
-  vi.mocked(readdirSync).mockReturnValue(data.map(({path}) => path) as any)
+  vi.mocked(readdirSync).mockReturnValue([...data].reverse().map(({path}) => path) as any)
   data.forEach(({run}) => vi.mocked(readFile).mockResolvedValueOnce(JSON.stringify(run) as any))
```

Comment:
Since replay reversed the list of filenames, there was a mismatch of filename to file content, previously not identified since we weren't changing the file content in any way.

The fix here is to apply a reverse- though a painful debugging session I finally realized that `reverse` reverses the array in place (?!?!?), so I'm making a new array to reverse to avoid causing undo chaos, especially in the one test that reuses the array after sending it to this method.

Other tests had compensated for this by reversing the file list in the expectation and relying on the fact that the only field which was changing was the filename (because of the random identifier). I've un-reversed those expectations so that they return in the order provided.

Reply:
Ah yeah, `reverse` is an old JS method and does it in place 😢

Reply:
If I ever get my hands on a time machine... 😆